catalyst-gateway/bin/src/db/event/config/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! Configuration query
+
+use jsonschema::BasicOutput;
+use key::ConfigKey;
+use serde_json::Value;
+use tracing::error;
+
+use crate::db::event::EventDB;
+
+pub(crate) mod key;
+
+/// Configuration struct
+pub(crate) struct Config {}
+
+/// SQL get configuration.
+const GET_CONFIG: &str = include_str!("sql/get.sql");
+/// SQL update if exist or else insert configuration.
+const UPSERT_CONFIG: &str = include_str!("sql/upsert.sql");
+
+impl Config {
+    /// Retrieve configuration based on the given `ConfigKey`.
+    ///
+    /// # Returns
+    ///
+    /// - A JSON value of the configuration, if not found, returns the default value.
+    /// - Error if the query fails.
+    pub(crate) async fn get(id: ConfigKey) -> anyhow::Result<Value> {
+        let (id1, id2, id3) = id.to_id();
+        let rows = EventDB::query(GET_CONFIG, &[&id1, &id2, &id3]).await?;
+
+        if let Some(row) = rows.first() {
+            let value: Value = row.get(0);
+            match id.validate(&value) {
+                BasicOutput::Valid(_) => Ok(value),
+                BasicOutput::Invalid(errors) => {
+                    // This should not happen; expecting the schema to be valid
+                    error!("Validate schema failed: {:?}", errors);
+                    Err(anyhow::anyhow!("Validate schema failed"))
```

Comment:
I would not fail here, just return the default if the value in the database is corrupted.